checker.go

```diff
@@ -203,6 +203,85 @@ func (c *matchesChecker) Check(got interface{}, args []interface{}, note func(ke
 	return BadCheckf("value is not a string or a fmt.Stringer")
 }
 
+// ErrorAs is a Checker checking that the error is or wraps the provided error
+// and if so, assigns it to the pointer. This is analogous to calling errors.As.
+//
+// For instance:
+//
+//     var pathError *os.PathError
+//     c.Assert(err, qt.ErrorAs, &pathError)
+//
+var ErrorAs Checker = &errorAsChecker{
```

Comment:
errors.As and errors.Is were only introduced in Go 1.13 and currently quicktest supports 1.11 and 1.12. Should we drop support for those versions or add these checkers in a file guarded by a go1.13 build tag?

Reply:
As mentioned elsewhere, it probably wouldn't take _too_ much code to create a pre-1.13 shim that would allow quicktest to work on pre-1.13 versions. Let me know if you'd like me to add this!